### ISSUE TITLE
language/enum/enhanced_enums_basic_test has multiple typos

### ISSUE THREAD

**Issue by scheglov** [dart-lang/sdk#48429]
@lrhn 
```
scheglov@scheglov-macbookpro4:~/Source/Dart/sdk.git/sdk (master)$ d-rat language/enum/enhanced_enums_basic_test
Test configuration:
    custom-configuration-1(architecture: x64, compiler: dart2analyzer, mode: release, runtime: none, system: mac)
Suites tested: language
Total: 1 tests
 * 0 tests will be skipped (0 skipped by design)
 * 0 tests are expected to be flaky but not crash
 * 0 tests are expected to flaky crash
 * 1 tests are expected to pass
 * 0 tests are expected to fail that we won't fix
 * 0 tests are expected to fail that we should fix
 * 0 tests are expected to crash that we should fix
 * 0 tests are allowed to timeout
 * 0 could not be categorized or are in multiple categories

2022-02-18 09:49:44.535856 Warning: Context messages in other files not currently supported.

FAILED: dart2analyzer-none release_x64 language/enum/enhanced_enums_basic_test
Expected: Pass
Actual: CompileTimeError

--- Command "dart2analyzer" (took 12.000481s):
DART_CONFIGURATION=ReleaseX64 sdk/bin/dartanalyzer -Dtest_runner.configuration=custom-configuration-1 --enable-experiment=enhanced-enums --ignore-unrecognized-flags --packages=/Users/scheglov/Source/Dart/sdk.git/sdk/.packages --format=json --no-hints /Users/scheglov/Source/Dart/sdk.git/sdk/tests/language/enum/enhanced_enums_basic_test.dart

unexpected analysis errors in enhanced_enums_basic_test.dart:
- Line 22, column 20: COMPILE_TIME_ERROR.UNDEFINED_IDENTIFIER
  Undefined name 'EnumPlainNoSemicolon'.

- Line 23, column 20: COMPILE_TIME_ERROR.UNDEFINED_IDENTIFIER
  Undefined name 'EnumPlainNoSemicolon'.

- Line 23, column 45: COMPILE_TIME_ERROR.UNDEFINED_IDENTIFIER
  Undefined name 'EnumPlainNoSemicolon'.

- Line 24, column 20: COMPILE_TIME_ERROR.UNDEFINED_IDENTIFIER
  Undefined name 'EnumPlainNoSemicolon'.

- Line 24, column 45: COMPILE_TIME_ERROR.UNDEFINED_IDENTIFIER
  Undefined name 'EnumPlainNoSemicolon'.

- Line 25, column 20: COMPILE_TIME_ERROR.UNDEFINED_IDENTIFIER
  Undefined name 'EnumPlainNoSemicolon'.

- Line 25, column 45: COMPILE_TIME_ERROR.UNDEFINED_IDENTIFIER
  Undefined name 'EnumPlainNoSemicolon'.

- Line 72, column 40: COMPILE_TIME_ERROR.UNDEFINED_ENUM_CONSTANT
  There's no constant named 'sFinalInit' in 'EnumAll'.

- Line 74, column 31: COMPILE_TIME_ERROR.UNDEFINED_ENUM_CONSTANT
  There's no constant named 'sLate' in 'EnumAll'.

- Line 75, column 11: COMPILE_TIME_ERROR.UNDEFINED_SETTER
  The setter 'sLate' isn't defined for the type 'EnumAll<S, T>'.

- Line 76, column 40: COMPILE_TIME_ERROR.UNDEFINED_ENUM_CONSTANT
  There's no constant named 'sLate' in 'EnumAll'.

- Line 77, column 11: COMPILE_TIME_ERROR.UNDEFINED_SETTER
  The setter 'sLate' isn't defined for the type 'EnumAll<S, T>'.

- Line 78, column 40: COMPILE_TIME_ERROR.UNDEFINED_ENUM_CONSTANT
  There's no constant named 'sLate' in 'EnumAll'.

- Line 93, column 32: COMPILE_TIME_ERROR.UNDEFINED_IDENTIFIER
  Undefined name 'TypedefAll'.

- Line 94, column 32: COMPILE_TIME_ERROR.UNDEFINED_IDENTIFIER
  Undefined name 'TypedefAll'.

- Line 122, column 29: COMPILE_TIME_ERROR.WRONG_NUMBER_OF_TYPE_ARGUMENTS_FUNCTION
  The function 'v1' is declared with 0 type parameters, but 1 type arguments were given.

- Line 142, column 10: COMPILE_TIME_ERROR.UNDEFINED_METHOD
  The method 'distinct' isn't defined for the type 'Expect'.

- Line 143, column 10: COMPILE_TIME_ERROR.UNDEFINED_METHOD
  The method 'distinct' isn't defined for the type 'Expect'.

- Line 271, column 30: COMPILE_TIME_ERROR.RETURN_OF_INVALID_TYPE
  A value of type 'void' can't be returned from the function 'notExtension' because it has a return type of 'String'.

- Line 283, column 3: COMPILE_TIME_ERROR.NON_CONST_GENERATIVE_ENUM_CONSTRUCTOR
  Generative enum constructors must be 'const'.

- Line 293, column 3: COMPILE_TIME_ERROR.NON_CONST_GENERATIVE_ENUM_CONSTRUCTOR
  Generative enum constructors must be 'const'.

- Line 303, column 3: COMPILE_TIME_ERROR.NON_CONST_GENERATIVE_ENUM_CONSTRUCTOR
  Generative enum constructors must be 'const'.

- Line 309, column 11: COMPILE_TIME_ERROR.ILLEGAL_NON_ABSTRACT_ENUM_INDEX
  A non-abstract instance member named 'index' can't be declared in a class that implements 'Enum'.

- Line 309, column 11: COMPILE_TIME_ERROR.INVALID_OVERRIDE
  'StringIndexEnum.index' ('num Function()') isn't a valid override of 'Enum.index' ('int Function()').

- Line 317, column 21: COMPILE_TIME_ERROR.VALUES_DECLARATION_IN_ENUM
  A member named 'values' can't be declared in an enum.

- Line 322, column 14: COMPILE_TIME_ERROR.VALUES_DECLARATION_IN_ENUM
  A member named 'values' can't be declared in an enum.

- Line 325, column 6: COMPILE_TIME_ERROR.ILLEGAL_ENUM_VALUES_INHERITANCE
  An instance member named 'values' can't be inherited from 'ValuesMixin' in a class that implements 'Enum'.

- Line 329, column 6: COMPILE_TIME_ERROR.ILLEGAL_ENUM_VALUES_INHERITANCE
  An instance member named 'values' can't be inherited from 'ValuesInterface' in a class that implements 'Enum'.

--- Re-run this test:
python3 tools/test.py -m release -c dart2analyzer language/enum/enhanced_enums_basic_test
[00:12 | 100% | +    0 | -    1]

=== 0 tests passed, 1 failed ===
```

**Comment by lrhn** [comment]
I think I've fixed those typos in https://dart-review.googlesource.com/c/sdk/+/231950
(I'll have to check if all of them are really fixed, and not just rewritten badly.)

**Comment by eernstg** [comment]
Note that https://dart-review.googlesource.com/c/sdk/+/233384 contains a handful of fixes that aren't in https://dart-review.googlesource.com/c/sdk/+/231950 as well.

**Comment by parlough** [comment]
Closing as resolved by db30a5f6702c8bc8dfe3c63ac7c662939d3f2351  and 46daf0fbd10fb3e06a690f11ce7abd2788bd3361. I also verified the file has no more analysis errors or typos :)
